### PR TITLE
api changes, thunks support added

### PR DESCRIPTION
**Pull request by jsdmc**
@erikras Breaking change is only for `mapStateToProps(key, state, ownProps)`. Now it's first argument is a multireducerKey passed to component. We can keep old signature and access `multireducerKey` from `ownProps`, but as you mentioned earlier, react-redux's `connect` makes some optimizations - thus it's better not to use `ownProps` if it's not used. Also now whole state passed to `mapStateToProps`. I think it's more flexible way. You can still grab needed state slice using `key` param.
@adailey14 Please take a look as there is a bunch of your changes)

### CODE REVIEW

**Comment by erikras** [comment]
To get the build to work on Node 0.12, please change
```js
action.type.includes(key)
```
to
```js
~action.type.indexOf(key)
```
Such a great js hack, that one. :-)

**Comment by jsdmc** [comment]
@erikras fixed. Relied on babel  here) Expected `includes` to work after transpiling. The same issue with `find` method of Arrays.

File: src/connectMultireducer.js [hunk 1]
```diff
@@ -28,22 +30,15 @@ export default function connectMultireducer(mapStateToProps, actions = {}) {
...
-              if (!multireducerKeys.filter(key => key === multireducerKey)[0]) {
-                throw new Error(`No state for multireducer key "${multireducerKey}". You initialized multireducer with "${multireducerKeys.join(', ')}".`);
-              }
-              const slice = state.multireducer[multireducerKey];
-              return mapStateToProps ? mapStateToProps(slice, ownProps) : slice;
-            },
-            multireducerBind(actions, multireducerKey)
+            wrapMapStateToProps(mapStateToProps, multireducerKey),
+            wrapMapDispatchToProps(mapDispatchToProps, multireducerKey),
+            ...rest
```
Comment (erikras) [inline src/connectMultireducer.js]: @jsdmc Somewhere in here, you lost the [BUT ONE IMPORTANT DIFFERENCE](https://github.com/erikras/multireducer#-mapstatetopropskey-state-ownprops--function-optional) logic that was slicing up the state to only the slice that the reducer wants. Does that make sense?

Comment (erikras) [inline src/connectMultireducer.js]: Or am I expected, as the user of the library, to slice my own state: `state.multireducer[key]` in `mapStateToProps`?

Comment (jsdmc) [inline src/connectMultireducer.js]: @erikras Oh, sorry if described it not clear. But yes you need to use that `key` to access state slice. Yeah, it's not that convenient as in previous version, but at the other hand you have access to app state.

Comment (erikras) [inline src/connectMultireducer.js]: Okay. Can you do another PR to fix the docs?

Comment (jsdmc) [inline src/connectMultireducer.js]: @erikras done

**Comment by danosaure** [comment]
This change kind of break encapsulation. I was using multireducer to not need to know where in the tree the "sub-state" is, now, it seems that I need to be aware of the whole state.
